dts/bindings/memory-controllers/st,stm32-ospi-psram.yaml

```diff
@@ -0,0 +1,149 @@
+# Copyright (c) 2025 ZAL Zentrum für Angewandte Luftfahrtforschung GmbH
+# SPDX-License-Identifier: Apache-2.0
+
+title: STM32 OSPI PSRAM Controller
+
+description: |
+  STM32 OSPI PSRAM memory controller.
+  Provides a communication interface allowing the microcontroller
+  to communicate with an external PSRAM memory via the OSPI peripheral.
+  The PSRAM can be configured as memory mapped and allow read/write
+  operations like an internal device.
+
+compatible: "st,stm32-ospi-psram"
+
+include: [base.yaml]
+
+properties:
+  reg:
+    required: true
+
+  size:
+    type: int
+    required: true
+    description: Memory size in bits
+
+  max-frequency:
+    type: int
+    required: true
+    description: Maximum clock frequency of device's OSPI interface in Hz
+
+  fixed-latency:
+    type: boolean
+    description: Read latency — fixed when set, variable otherwise.
+
+  drive-strength:
+    type: int
+    default: 0
+    description: |
+      Drive strength code:
+      - 0: Full
+      - 1: Half
+      - 2: 1/4
+      - 3: 1/8
+    enum:
+      - 0
+      - 1
+      - 2
+      - 3
+
+  read-latency:
+    type: int
+    default: 5
+    description: |
+      Read latency code
+      - 0: LCx2 3 / Fmax 66 MHz
+      - 1: LCx2 4 / Fmax 104 MHz
+      - 2: LCx2 5 / Fmax 133 MHz
+      - 3: LCx2 6 / Fmax 133 MHz
+      - 4: LCx2 7 / Fmax 133 MHz
+      - 5: LCx2 8 / Fmax 133 MHz
+    enum:
+      - 0
+      - 1
+      - 2
+      - 3
+      - 4
+      - 5
+
+  write-latency:
+    type: int
+    default: 2
+    description: |
+      Write latency code:
+      - 0: LC 3 / Fmax 66 MHz
+      - 1: LC 4 / Fmax 104 MHz
+      - 2: LC 5 / Fmax 133 MHz
+      - 3: LC 6 / Fmax 133 MHz
+      - 4: LC 7 / Fmax 133 MHz
+      - 5: LC 8 / Fmax 133 MHz
+    enum:
+      - 0
+      - 1
+      - 2
+      - 3
+      - 4
+      - 5
+
+  refresh-rate:
```

Comment:
The property name is a bit ambiguous. How about `adaptive-refresh-rate` or `auto-adjust-refresh-rate`?